Realm/RLMSyncUser.h

```diff
@@ -111,6 +112,13 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, readonly) RLMSyncUserState state;
 
+/**
+ Returns the default configuration for the user. The default configuration points to the default
+ query-based Realm on the server the user authenticated against.
+*/
+@property (nonatomic, readonly) RLMRealmConfiguration *defaultConfiguration NS_REFINED_FOR_SWIFT;
```

Comment:
This should just be a method named `configuration`.

Reply:
Shouldn't it just be deleted if we rename the `createConfiguration` to `configuration` ?

Reply:
In obj-c we'd need both `configuration` and `configurationWithURL:` because there's no default parameters, but in Swift they can be a single function.

Reply:
One concern I have about this is that `automatic()` currently returns `Realm.Configuration`. If we want to still have a sensible default is it then common in Swift to have a property named `configuration` that is default and a method named `configuration()` that can be used to customize it?

Reply:
Objc-c:
```
- (RLMRealmConfiguration *)configuration;
- (RLMRealmConfiguration *)configurationWithURL:(NSURL *)realmURL;
```

The developer would write `user.configuration` or `[user configuration]` to get the default Realm, or `[user configurationWithURL:realmURL]` to set the URL.

Swift:
```
public func configuration(realmURL: URL? = nil, fullSynchronization: Bool = false, enableSSLValidation: Bool = true, urlPrefix: String? = nil) -> Realm.Configuration
```

The developer would call `user.configuration()` for the default realm, or `user.configuration(realmURL: url)` to customize it. The property would just let them drop the `()`, so there's no real point.